src/libpmem2/extent_linux.c

```diff
@@ -35,23 +38,38 @@ os_extents_common(const char *path, struct extents *exts,
 		return -1;
 	}
 
-	enum file_type type = util_fd_get_type(fd);
-	if (type < 0)
+	os_stat_t st;
+	if (os_fstat(fd, &st) < 0) {
+		ERR("!fstat %d", fd);
 		goto error_close;
+	}
 
-	struct stat st;
-	if (fstat(fd, &st) < 0) {
-		ERR("!fstat %d", fd);
+	enum pmem2_file_type pmem2_type;
+
+	int ret = pmem2_get_type_from_stat(&st, &pmem2_type);
+	if (ret) {
+		errno = pmem2_err_to_errno(ret);
 		goto error_close;
 	}
 
+	switch (pmem2_type) {
+		case PMEM2_FTYPE_REG:
+		case PMEM2_FTYPE_DIR:
+		case PMEM2_FTYPE_DEVDAX:
+			break;
+		default:
+			ASSERTinfo(0,
+				"unhandled file type in pmem2_get_type_from_stat");
+			return -1;
+	};
```

Comment:
Pasting this switch everywhere util_fd_get_type was used doesn't look good - try to integrate it with the surrounding function. E.g. in this function DEVDAX is handled later, DIR doesn't make sense, so it should error out. The only thing left is REG and out-of-switch value, so ASSERTing that type == REG is enough.